docs/caching/index.md

```diff
@@ -0,0 +1,104 @@
+# App-Level Context Caching
+
+The Application Development Kit (ADK) provides an app-level context caching mechanism to improve performance and reduce costs. By caching the static parts of your agent's instructions, you can significantly speed up responses and lower the number of tokens sent to the model for each request.
+
+This document explains how to configure and use this feature.
+
+## `ContextCacheConfig`
+
+To enable context caching, you need to provide a `ContextCacheConfig` object when you create your `App`. This configuration is applied to all LLM agents within the app.
+
+`ContextCacheConfig` has the following fields:
+
+- `min_tokens` (int): The minimum number of tokens required in a request to enable caching. This is useful for avoiding the overhead of caching for very small requests where the performance benefit would be negligible. Defaults to `0`.
+- `ttl_seconds` (int): The time-to-live (TTL) for the cache in seconds. This determines how long the cached content will be stored before it's refreshed. Defaults to `1800` (30 minutes).
+- `cache_intervals` (int): The maximum number of times the same cached content can be used before it's refreshed. This allows you to control how frequently the cache is updated, even if the TTL has not expired. Defaults to `10`.
+
+## `static_instruction` vs. `instruction`
+
+To use context caching, you need to separate the static and dynamic parts of your agent's instructions.
+
+- `LlmAgent.static_instruction`: Use this field for the static, unchanging parts of your instructions. This is the content that will be cached. It should be a `google.genai.types.Content` object.
+- `LlmAgent.instruction`: Use this field for the dynamic parts of your instructions that can change between requests. This can be a string or a function that returns a string.
+
+When you use `static_instruction`, the ADK will automatically handle caching this content for you based on the `ContextCacheConfig` you provide.
+
+## Code Example
+
+Here's an example of how to use app-level context caching with a digital pet agent named "Bingo." Bingo's core personality is defined in `static_instruction`, while its mood, which changes based on when it was last fed, is provided in the dynamic `instruction`.
+
+```python
```

Comment:
This example is OK, but a bit too elaborate, and therefore. Hard to follow.

re-write it, to compress the size of this code to make it easier to understand